app/controllers/assignments_controller.rb

```diff
@@ -211,7 +216,8 @@ def assignment_params
                                ],
                                graders_attributes: [
                                  :avail_score, :upload_file, :params,
-                                 :type, :id, :_destroy
+                                 :type, :id, :_destroy, :errors_to_show, :test_class,
```

Comment:
This is a bit sketchy, since it marks upload_by_user_id as part of the network interface.

This shouldn't be terminal here, since this value doesn't do anything and the mutation methods are all restricted to staff or higher, but it's not good.